packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -207,6 +230,40 @@ export class Cluster extends Resource implements ICluster {
     }
   }
 
+  private renderConfiguration() : CfnCluster.ClusterConfigurationProperty | undefined {
```

Comment:
We try to avoid using type unions for sake of JSII compatibility. You should be able to do something like:

```
configuration: this._executeCommandConfiguration && this.renderExecuteCommandConfiguration()
```
in line 192 to set undefined as needed, and then `renderConfiguration()` won't be needed.

Reply:
Yup, that was super neat! Thanks!